examples/oap/Makefile

```diff
@@ -53,8 +53,14 @@ $(BOOTLOADER): FORCE
 $(GOLDEN_IMAGE): FORCE
 	$(MAKE) -C $(dir $@) TARGET=$(TARGET) BOARD=$(BOARD) $(notdir $@)
 
+OS_VERS := $(shell lsb_release -a 2>/dev/null | grep Description | awk '{ print $$2 "-" $$3 }')
```

Comment:
👎 

This won't run on macOS, so you should only run `lsb_release` after you are confident you are on Linux.

A good way to do OS detection with make can be found here: https://stackoverflow.com/a/12099167

I would then use a `PYTHON` variable and set it accordingly inside your OS detection, falling back to `python` if unset.

```
PYTHON := python

$(PYTHON) $(GENERATE_METADATA) -o $@ -O $(METADATA_OFFSET) $<
```

One could of course also default to `python3` these days...

Reply:
Hmmm OK. I just noticed the `2>/dev/null` part.

I wonder whether what we should do instead is, check if python3 exists and fall back to python if not. Instead of worrying about specific OS versions

Reply:
I've just set-up a Ubuntu 20.04 VM and it only has "python3" by default, and fails on "python " line. I assume it might be an issue for others on Ubuntu 20.04 to use the patch 

(Btw, I think OAP work looks more than good, I hope it'd be merged into the development branch soon)

maybe something like this would be better on Linux side to test if the OS has python3, than testing the OS version https://stackoverflow.com/a/54844990

Reply:
```
PYTHON = $(shell type -P python3)
PYTHON ?= python
```

I am not sure how something like this might behave on Windows. Perhaps there should be something like:

```make
ifeq ($(OS),Windows_NT)
  # Do something for windows e.g.
  PYTHON = python
else
  PYTHON := $(shell type -P python3)
endif

PYTHON ?= python
```

Reply:
Perhaps even better

```make
ifneq ($(OS),Windows_NT)
  PYTHON := $(shell type -P python3)
endif

PYTHON ?= python
```